ddtrace/profiling/profiler.py

```diff
@@ -208,9 +213,14 @@ def __attrs_post_init__(self):
         exporters = self._build_default_exporters()
 
         if exporters:
-            self._scheduler = scheduler.Scheduler(
-                recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
-            )
+            if self._lambda_function_name:
+                self._scheduler = serverless_scheduler.ServerlessScheduler(
+                    recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
+                )
+            else:
+                self._scheduler = scheduler.Scheduler(
+                    recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
+                )
```

Comment:
```suggestion
            if self._lambda_function_name:
                scheduler_class = serverless_scheduler.ServerlessScheduler
            else:
                scheduler_class = scheduler.Scheduler 
            self._scheduler = scheduler_class(
                    recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
                )
```

Reply:
This throws a type error via mypy. I've added a Union type in the attribute declaration (ha, just realized that's what `attr.ib` means. Cute), but it still errs: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/19743/workflows/ef8b783d-20e6-4661-a796-fcf0c9d72718/jobs/1335294